### PR TITLE
feat(chat): Pending messages

### PR DESCRIPTION
**Pull request by Flemmli97**
### What this PR does 📖

- Implements pending messages. When sending a ghost message will appear till the chat has been sent to warp.
- When sending files with attachments a progress bar will also be shown. If an error occurs during upload a button to delete that file will also appear
- Implements processing the stream of attachments (aka actually handling file uploads) which was previously not done yet (Untested yet though as its some ui stuff to actually upload files is not implemented atm)

### Note
- For normal messages the ghost message will appear very short when testing locally

### CODE REVIEW

**Review by InfamousVague — COMMENTED** [review]
Looks good but there is quite a bit of reused code from the standard version of these messages, instead what if we implemented these changes on the standard embeds, messages etc and just apply a "pending" property to apply the pending styles and other changes.

**Comment by Flemmli97** [comment]
> Looks good but there is quite a bit of reused code from the standard version of these messages, instead what if we implemented these changes on the standard embeds, messages etc and just apply a "pending" property to apply the pending styles and other changes.

i though about that but i felt like both the differences and similarities are pretty equal in amount and for organization sake split it from it

**Comment by InfamousVague** [comment]
Hmm, maybe we can change it so that the only change for the pending message is an opacity change, this way we can just pass the `pending` prop and modify the opacity on the message and change the context options (in the future). I am just worried that this will become annoying to maintain since we have two near identical components with just slight changes.

File: src/lib/wasm/RaygunStore.ts [hunk 1]
```diff
@@ -535,6 +539,77 @@ class RaygunStore {
...
+     * Create a handler for attachment results that uploads the file to chat and updates pending message attachments
+     * TODO: verify it works as we dont have a way to upload files yet
+     */
+    private async createFileAttachHandler(conversationId: string, upload: wasm.AttachmentResult) {
+        let listener = {
+            [Symbol.asyncIterator]() {
+                return upload
+            },
+        }
+        let cancelled = false
```
[the remark this answers is not in the document]
Comment (schwindy) [reply src/lib/wasm/RaygunStore.ts:552]: That's not true, mis-spell bot :monkey:

File: src/lib/wasm/RaygunStore.ts [hunk 2]
```diff
@@ -535,6 +539,77 @@ class RaygunStore {
...
+                                        current.done = true
+                                        break
+                                    }
+                                    case "ProgressFailed": {
+                                        current.size = progress.values["last_size"]
+                                        current.error = progress.values["error"]
+                                        break
+                                    }
+                                }
+                                current
```
Comment (schwindy) [inline src/lib/wasm/RaygunStore.ts:583]: This is line intended? I don't believe it does anything.

Comment (Flemmli97) [reply src/lib/wasm/RaygunStore.ts:583]: my mistake. it shouldve returned

**Pull request merged**